pkg/types/validation/installconfig.go

```diff
@@ -699,3 +703,31 @@ func validateFIPSconfig(c *types.InstallConfig) field.ErrorList {
 	}
 	return allErrs
 }
+
+// validateCapabilities checks if additional, optional openshift components are specified in the
+// install-config to be included in the installation
+func validateCapabilities(c *types.Capabilities, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if c.BaselineCapabilitySet == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("baselineCapabilitySet"), "baseline capability set required"))
+
+		return allErrs
+	}
+	capabilitySet := make(map[configv1.ClusterVersionCapability]bool)
+	// Check to see if the additionalEnabledCapabilities is defined in openshift/api
+	if availableAdditionalCapabilties, found := configv1.ClusterVersionCapabilitySets[c.BaselineCapabilitySet]; found {
+		// Populate the available capability set map for the chosen baselineCapablitySet
+		for _, capability := range availableAdditionalCapabilties {
+			capabilitySet[capability] = true
+		}
+		// Check for additionalEnabledCapabilities input by user
+		for _, inputCapability := range c.AdditionalEnabledCapabilities {
+			if capabilitySet[inputCapability] != true {
+				allErrs = append(allErrs, field.Invalid(fldPath, c.AdditionalEnabledCapabilities, fmt.Sprintf("unknown additional enabled capability specified")))
+			}
+		}
```

Comment:
nit: Just a thought but I think using sets would be easier here. I'm fine either way
```
// Check to see if the additionalEnabledCapabilities is defined in openshift/api
if availableCapabilities, found := configv1.ClusterVersionCapabilitySets[c.BaselineCapabilitySet]; found {
	// Populate the available capability set map for the chosen baselineCapablitySet
	capabilitySet := sets.NewString(availableCapabilities...)
	// Check for additionalEnabledCapabilities input by user
	missingCapability := sets.NewString(c.AdditionalEnabledCapabilities...).Difference(capabilitySet)
	for _, missing := range missingCapability.List() {
		allErrs = append(allErrs, field.Invalid(fldPath, missing, fmt.Sprintf("unknown additional enabled capability specified")))
	}
}
```

Reply:
Looks a lot more elegant than using a map. Will change. Thanks for the tip.